src/MarkwhenView.ts

```diff
@@ -100,21 +99,13 @@ export class MarkwhenView extends TextFileView {
 		});
 
 		this.registerDomEvent(window, 'message', async (e) => {
-			if (e.source == this.iframe.contentWindow) {
-				// console.log(e);
-				// handle received global message here
-				/* Sample e.data
-					{
-						"type": "setHoveringPath",
-						"request": true,
-						"id": "markwhen_WIPSNWsi1BPsoc5PIRwRTMicRS3xVGGX"
-					}
-					{
-						"type": "appState",
-						"response": true,
-						"id": "markwhen_CkNiTDyiNpVq6ZAjbsrsDEn8PgL9ERbn"
-					}
-				*/
+			if (e.source == this.iframe.contentWindow && e.data.request) {
+				if (e.data.type === 'newEvent') {
+					const newEventString = `\n${dateRangeToString(
+						toDateRange(e.data.params.dateRangeIso)
+					)}: new event`;
+          this.setViewData(this.data + newEventString, false)
```

Comment:
I don't know if this is the right way to update the document but that's what I'm trying to do here